### PR TITLE
Decrypt support

### PR DESCRIPTION
**Pull request by pitbulk**
Initial decrypt support. Related to #9

### CODE REVIEW

**Comment by pitbulk** [comment]
@Lordnibbler what do you think about this PR?

File: lib/onelogin/ruby-saml/response.rb [hunk 1]
```diff
@@ -184,8 +195,8 @@ def validate_response_state(soft = true)
       end
 
       def xpath_first_from_signed_assertion(subelt=nil)
-        node = REXML::XPath.first(document, "/p:Response/a:Assertion[@ID='#{document.signed_element_id}']#{subelt}", { "p" => PROTOCOL, "a" => ASSERTION })
-        node ||= REXML::XPath.first(document, "/p:Response[@ID='#{document.signed_element_id}']/a:Assertion#{subelt}", { "p" => PROTOCOL, "a" => ASSERTION })
+        node = REXML::XPath.first(assertion_document, "/samlp:Response/saml:Assertion[@ID='#{assertion_document.signed_element_id}']#{subelt}", { "samlp" => PROTOCOL, "saml" => ASSERTION, 'ds' => DSIG })
```
Comment (Lordnibbler) [inline lib/onelogin/ruby-saml/response.rb]: this change does not incorporate the security fix from 0.8.2

https://github.com/onelogin/ruby-saml/blob/master/lib/onelogin/ruby-saml/response.rb#L186-L200

**Comment by Lordnibbler** [comment]
I'd recommend we add documentation and tests for each new method added in `response.rb`

**Comment by pitbulk** [comment]
Replaced by #241

**Pull request closed without merging**